src/background/messenger.ts

```diff
@@ -19,36 +19,45 @@ export interface ExtensionAdapter {
 }
 
 export default class Messenger {
-    private reporters: Set<(info: ExtensionData) => void>;
     private adapter: ExtensionAdapter;
+    private changeListenerCount: number;
 
     constructor(adapter: ExtensionAdapter) {
-        this.reporters = new Set();
         this.adapter = adapter;
-        chrome.runtime.onConnect.addListener((port) => {
-            if (port.name === 'ui') {
-                port.onMessage.addListener(async (message) => await this.onUIMessage(port, message));
+        this.changeListenerCount = 0;
+        chrome.runtime.onMessage.addListener((message: Message, sender: any, sendResponse: (any) => void) => {
+            if (message.name === 'ui') {
+                this.onUIMessage(message, sendResponse);
                 this.adapter.onPopupOpen();
+                return ([
```

Comment:
The type defintion of `ExtensionMessageEvent` don't expect any return value:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/6f27c44c9a0040d4d86f3a17738fbf39fd616ed2/types/chrome/index.d.ts#L6574

Any reason why this is included?

Reply:
This is crucial for letting `onUIMessage` to use `sendResponse`. Docs say [this](https://developer.chrome.com/docs/extensions/reference/runtime/#event-onMessage):

> This function becomes invalid when the event listener returns, **unless you return true** from the event listener to indicate you wish to send a response asynchronously (this will keep the message channel open to the other end until `sendResponse` is called).

Reply:
Alternatively, we could move all logic directly into `onUIMessage` (since it's just a few LoC) and write just `chrome.runtime.onMessage.addListener(this.onUIMessage)`. Then `return` statement would be unnecessary.

Reply:
Heh, that's good to know. That's worth opening up a issue in the type definitions repo.

Reply:
That sounds like a better idea as it might come off confusing why to return something - Especially on this behavior with chromium